rust/tests/read_delta_test.rs

```diff
@@ -527,6 +528,40 @@ async fn test_poll_table_commits() {
     assert!(matches!(peek, PeekCommit::UpToDate));
 }
 
+#[tokio::test]
```

Comment:
@rtyler Here I'm testing that when buffering > 1 the processing is faster. Is there a nice way to patch the `get` method to insert a delay there, so that the test would work reliably? And do you think this kind of slightly stochastic test is OK or necessary to have? Or is it in the benchmark territory?

Reply:
The simplest path would be to implement a wrapper struct which implements `ObjectStore` and then implement the delay you seek in its `get` implementation.

This test would need to be changed to construct the table in a different fashion, such as with the table builder .